### PR TITLE
Version 2.0

### PR DESCRIPTION
**Pull request by jhawthorn**
This removes almost all of the functionality from the extension in favour of the new functionality which is built-in to Solidus itself.

This proposed version 2.0 of solidus_i18n provides nothing other than locale files, doesn't require deface, and doesn't do any `class_eval` or module prepends.

Functionality moved to Solidus itself:
* Frontend locale selector  solidusio/solidus#2683
* Per-store locales  solidusio/solidus#2674
* LocaleController override solidusio/solidus#2626
* `set_user_language` behaviour https://github.com/solidusio/solidus/pull/2559/commits/568ca0b584d720ebcac6211001262fb77a4a0775

Removed functionality:
* Remove SolidusI18n::Config (this shouldn't exist now that there are per-store locales)
* Remove weird country name magic (we want to remove this anyways, see #108, #106)
* Remove routing filter (this can be added by users, and should be recommended in the README)

TODO:
* [x] Upgrade instructions/changelog
* [x] README instructions for routing-filter gem
* [x] Remove rails-i18n and kaminari-i18n
* [x] Remove specs

### CODE REVIEW

**Comment by tvdeyen** [comment]
Tested while working on https://github.com/solidusio/solidus/pull/2684. Working great so far 👍 and fixes lots of bugs while switching admin locales. We should drop < 2.6 support IMO. There is still a version working for < 2.6 and translations files can easily be updated in the store application (this is recommended anyway)

File: README.md [hunk 1]
```diff
@@ -20,42 +20,36 @@ translation file into your application by following the
...
-Run `bundle install`
+## Locale in URL
 
-You can use the generator to install migrations and append solidus_i18n assets to
-your app solidus manifest file.
+Older versions of solidus_i18n included the routing-filter gem and configured routes to include the locale in the URL.
+This is still supported (maybe even recommended) but requires some additional configuration.
 
-    bin/rails g solidus_i18n:install
+1. Add gem to your `Gemfile`, then run `bundle install`
```
Comment (tvdeyen) [inline README.md]: Add _this_ gem to your ...

**Comment by tvdeyen** [comment]
We should remove `rails-i18n` and `kaminari-i18n` dependencies as well and encourage people to copy over the locale files they actually need over into their host app. I doubt anybody is using all locales available in their stores. Loading so many locales all the time only bloats their apps.

File: solidus_i18n.gemspec [hunk 2]
```diff
@@ -23,13 +23,10 @@ Gem::Specification.new do |s|
...
-  s.add_runtime_dependency 'i18n_data', '~> 0.7.0'
   s.add_runtime_dependency 'rails-i18n', ['>= 4.0.1', '< 6']
   s.add_runtime_dependency 'kaminari-i18n', '~> 0.5.0'
-  s.add_runtime_dependency 'routing-filter', '~> 0.6.0'
   s.add_runtime_dependency 'solidus_core', ['>= 1.1', '< 3']
   s.add_runtime_dependency 'solidus_support'
-  s.add_runtime_dependency 'deface', '~> 1.0'
 
   s.add_development_dependency 'byebug'
   s.add_development_dependency 'capybara', '~> 2.17'
```
Comment (tvdeyen) [inline solidus_i18n.gemspec]: Do we still have tests left, then?

Comment (jhawthorn) [reply solidus_i18n.gemspec]: I need to remove them still. I'd like some extremely simple tests (basically, verify YAML is valid). We won't need capybara

**Comment by jhawthorn** [comment]
This is ready to go :shipit:

I'm not sure what we should do to support versions < 2.6. Maybe a `solidus_i18n_backports` gem?

File: solidus_i18n.gemspec [hunk 3]
```diff
@@ -23,21 +23,9 @@ Gem::Specification.new do |s|
 
   s.has_rdoc = false
 
-  s.add_runtime_dependency 'i18n_data', '~> 0.7.0'
-  s.add_runtime_dependency 'rails-i18n', ['>= 4.0.1', '< 6']
-  s.add_runtime_dependency 'kaminari-i18n', '~> 0.5.0'
-  s.add_runtime_dependency 'routing-filter', '~> 0.6.0'
   s.add_runtime_dependency 'solidus_core', ['>= 1.1', '< 3']
   s.add_runtime_dependency 'solidus_support'
```
Comment (tvdeyen) [inline solidus_i18n.gemspec]: Do we still need `solidus_support`? Couldn’t find any usages anymore.

Comment (jhawthorn) [reply solidus_i18n.gemspec]: Nope!

**Pull request merged**